### ISSUE TITLE
IPython4 bash magic ! with {} does not work with Python 3.5.1

### ISSUE THREAD

**Issue by boeddeker** [ipython/ipython#9120]
Hello,

under http://bugs.python.org/issue26062 I have reported this bug to python. They told me that I ask ipython for a fix.

in python 3.5.1 the function Formatter._vformat in lib/python3.5/string.py has changed. Because of that the following is not working:
>In [1]: a = 3
>In [2]: !echo {a} # expected output: 3
>{a}
>In [3]: get_ipython().system('echo {a}') # expected output: 3
>{a}
>In [4]: get_ipython().var_expand('echo {a}', depth=1) # expected output: echo 3
>Out[4]: 'echo {a}'

Functions that will be called:
lib/python3.5/string.py :  vformat
lib/python3.5/site-packages/IPython/utils/text.py : _vformat

Python 3.5.0:

~~~
def _vformat( ..., auto_arg_index=0):
    ...
    format_spec = self._vformat( ..., auto_arg_index=auto_arg_index)
    ...
    return ''.join(result)
~~~

Python 3.5.1:

~~~ 
def _vformat( ..., auto_arg_index=0):
   ...
   format_spec, auto_arg_index = self._vformat( ..., auto_arg_index=auto_arg_index)
   ...
   return ''.join(result), auto_arg_index
~~~

Best regards
Christoph

**Comment by takluyver** [comment]
Thanks, I think we already fixed that in #8985, so it's probably just waiting for a new release.

**Comment by boeddeker** [comment]
Sorry, it's already fixed in IPython 4.0.1, I had the old version 4.0.0
